Helm integration (#35)
TheGreenMachine/GreenScout-Backend: docs/SelfHosting.md

> # Kuberneties Basicks

Reviewer: ```suggestion
# Kubernetes Basics
```

> Kuberneties is a platform that is desinged to be scalable so it is a little difrent then your run of a mill docker stack. the main difrance is the way kuberneties handles work loads. In a kuberneties cluster you should have both A Controll Plane, and (a or many) Worker Nodes. The controll plane is what handles the dispatching of the work and the worker nodes dose all the work.

Reviewer: ```suggestion
Kubernetes (k8s) is a platform that is designed to be scalable so it is a little diffrent than your run of the mill docker stack. The main difference is the way k8s handles workloads. A k8s cluster has a set of nodes that comprise the "control plane", which handles the placement or "scheduling" of a workload, and one or more "worker" nodes that actually run the workloads.
```

> # machine infraestructure

Reviewer: ```suggestion
# Example Infrastructure
```

> I (Landon) virtualise all of my kuberneties nodes, at the time i was deploying this app i had only two worker nodes and one controll plane do whatever you plese just keep in mind the more machines u have the better u can push out more replicas for smother sailing. Now when setting up our vm inferstructure i recomend using IAC(infrastructure as code) i have my homelabs infrastructure published down below to use as an example.

Reviewer: ```suggestion
I (Landon) run all of my k8s nodes as VMs. At the time I was deploying this app I had only two worker nodes and one control plane node. Do whatever you please just keep in mind having more nodes increases the resiliency of your services. When setting up our VM infrastructure I recommend using [IAC (infrastructure as code)](https://en.wikipedia.org/wiki/Infrastructure_as_code) tools to manage them. I have my home lab's infrastructure published down below to use as an example.
```

[Skoxie-iac Repo](https://codeberg.org/Woxie/skoxie-iac)

> # Where do i get the Images?

Reviewer: ```suggestion
# Where do I get the images?
```

> U(points at u) future me.. will have to push a docker image to docker hub or your own docker registry and then use that with a helm chart. I use Artifact hub to publish my completed charts so that i can pull them into my argo cd infrastructure

Reviewer: ```suggestion
U (points at you), future me... will have to push a docker image to docker hub or your own docker registry, and then use that with a helm chart. I use Artifact hub to publish my completed charts so that i can pull them into my [ArgoCD](https://argo-cd.readthedocs.io/en/stable/) infrastructure
```

> # the helm repo
> the helm chart repo is hosted on a github page evrytime you want to update the charts plese update the vershion number and push to main thanks

Reviewer: ```suggestion
# The Helm repo
The Helm chart repo is published using GitHub Pages. Everytime you want to update the charts please update the version number and submit a PR to main, thanks.
```
